client/src/index.tsx

```diff
@@ -1,9 +1,18 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 import reportWebVitals from './reportWebVitals';
 import App from './App';
+import { AlertProvider } from './util/context/AlertContext';
+
+const container = document.getElementById('root');
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const root = createRoot(container!); // createRoot(container!) if you use TypeScript
```

Comment:
What exactly is the createRoot used for here?

Reply:
Its a React 18 change (https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis). Honestly not sure what is does but the code comes from docs so I trust